packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/SchemaDrivenAnnotationsEditor.integration.test.tsx

```diff
@@ -348,7 +347,85 @@ describe('SchemaDrivenAnnotationEditor tests', () => {
       ...schemaHandlers,
       successfulUpdateHandler,
     )
-    await renderComponent()
+    const { container } = await renderComponent()
+    await screen.findByText('requires scientific annotations', { exact: false })
+
+    await waitFor(() => {
+      const countryInput = container.querySelector(
+        '#root_country',
+      ) as HTMLInputElement
+      expect(countryInput.value).toBe('USA')
+    })
+
+    // Saving the form should maintain the existing annotations
+    await clickSave()
+    await waitFor(() =>
+      expect(updatedJsonCaptor).toHaveBeenCalledWith(
+        expect.objectContaining({ country: 'USA', state: 'Washington' }),
+      ),
+    )
+  })
+
+  it('Reloads the form data properly after remounting the component (SWC-6486)', async () => {
```

Comment:
This new test failed before adding changes, so I think this is a valid reproduction